components/expenses/PayoutMethodForm.js

```diff
@@ -52,6 +52,34 @@ export const validatePayoutMethod = payoutMethod => {
     if (!payoutMethod.data.accountHolderName) {
       set(errors, 'data.accountHolderName', createError(ERROR.FORM_FIELD_REQUIRED));
     }
+    /*
+     * Validate the account holder name against the legal name. Following cases are considered a match,
+     *
+     * 1) Punctuation are ignored; "Evil Corp, Inc" and "Evil Corp, Inc." are considered a match.
+     * 2) Accents are ignored; "François" and "Francois" are considered a match.
+     * 3) The first name and last name order is ignored; "Benjamin Piouffle" and "Piouffle Benjamin" is considered a match.
+     */
+    if (payoutMethod.data.accountHolderName && legalName) {
+      const namesArray = legalName.split(' ');
+      let legalNameReversed;
+      if (namesArray.length === 2) {
+        const firstName = namesArray[0];
+        const lastName = namesArray[1];
+        legalNameReversed = `${lastName} ${firstName}`;
+      }
+      if (
+        payoutMethod.data.accountHolderName.localeCompare(legalName, undefined, {
+          sensitivity: 'base',
+          ignorePunctuation: true,
+        }) &&
+        payoutMethod.data.accountHolderName.localeCompare(legalNameReversed, undefined, {
+          sensitivity: 'base',
+          ignorePunctuation: true,
+        })
+      ) {
```

Comment:
This code is complex, let's extract it to a generic helper to simplify the code here. I suggest creating a new `compareNames` helper in `lib/collective.lib.js`.